albinos-cli/source/app.d

```diff
@@ -0,0 +1,86 @@
+import std.stdio;
+import std.string;
+import std.socket : UnixAddress, Socket, SocketType, AddressFamily, SocketOSException;
+import std.uni : isWhite;
+import std.process : executeShell;
+
+///
+class Client
+{
+	///
+	this(string socket_path)
+	{
+		this.address_ = new UnixAddress(socket_path);
+		this.socket_ = new Socket(AddressFamily.UNIX, SocketType.STREAM);
+		socket_.connect(this.address_);
+	}
+
+	private UnixAddress address_;
+	private Socket socket_;
+}
+
+///
+class CLI
+{
+	this(string socket_path)
+	{
+		try
+		{
+			client_ = new Client(socket_path);
+		}
+		catch (SocketOSException e)
+		{
+			writeln(stderr, e.msg);
+			is_running_ = false;
+		}
+	}
+	///
+	public void run_program()
+	{
+		if (!is_running_)
+			return;
+		write("> ");
+		string line;
+		while (((line = stdin.readln().strip("\n \t")) !is null))
+		{
+			if (line == "exit")
+			{
+				break;
+			}
+			auto splited_line = line.split!isWhite;
+			//! Command with argument
+			if (splited_line.length >= 2)
```

Comment:
Why do you need to handle both cases separately? This seems like it will duplicate a lot of code in the future.

Reply:
For example for commands like `help` this will not be lot of duplicate honnestly

Reply:
Yes, but it is easier to make help work in the general case (check there are no arguments) than the opposite. Besides, help could typically also take arguments to explain what the different commands are.

Reply:
Okok